### DISCUSSION TITLE
SSH into Codespaces or separate terminal window

### DISCUSSION THREAD

**Discussion by delucca** [community/community#3651] [Codespaces] [answered]
Hi everyone!

I'm using Github Codespaces as my primary IDE for a week now. Everything is working great, but I would like to know if I can detach the IDE integrated terminal for a different monitor, or connect easily with my Codespace through SSH.

I would like to do so because I have 2 monitors, and it is a waste of space using a small integrated terminal since I could have it on a separate window.

Thanks in advance!

**Comment by 2percentsilk** [comment, marked as answer]
HI @delucca,

We do have a way for you to SSH to a codespace detailed in [these docs](https://github.com/microsoft/vscode-dev-containers/blob/master/script-library/docs/sshd.md#usage-when-this-script-is-already-installed-in-an-image)

Give it a shot and let us know how it goes?

And out of curiosity, are you primarily using your terminal with vim or just generally like the terminal to be larger than the integrated one in VS Code?

**Reply by delucca** [reply]
> HI @delucca,
> 
> We do have a way for you to SSH to a codespace detailed in [these docs](https://github.com/microsoft/vscode-dev-containers/blob/master/script-library/docs/sshd.md#usage-when-this-script-is-already-installed-in-an-image)
> 
> Give it a shot and let us know how it goes?
> 
> And out of curiosity, are you primarily using your terminal with vim or just generally like the terminal to be larger than the integrated one in VS Code?

Thanks!

I skipped that section from the docs, I'm going to read it and give it a try!

Regarding your question, I'm trying to use Github Codespaces as my primary/only IDE. I was using Atom previously, but the lack of support/features was killing me.

In my local workflow, it is common to switch between terminal/editor. I have 3 monitors, so usually, I keep the terminal in my left monitor, the IDE in the middle, and the website (or the presentation view I'm working on) in the right.

That is extremely handy for me because I can switch to my terminal, debug/explore something, and (without pressing any keys) look to the other side and inspect it directly in the browser.

So, that is what I'm trying to accomplish. It really does not matter if I'm using the terminal in the browser (because I've installed all my dotfiles with the amazing dotfiles feature that Github Codespaces have). That's why I'm considering doing an SSH connection with the container directly.

**Reply by 2percentsilk** [reply]
Got it! Thanks for the quick and thorough explanation! That really helps out with your use case (and helps me think about how we can make this easier for you in the future!) 😄

**Reply by delucca** [reply]
> Got it! Thanks for the quick and thorough explanation! That really helps out with your use case (and helps me think about how we can make this easier for you in the future!) smile

Glad I could help :)

Regarding the link you shared. It works also in Github Codespaces directly on the browser? Or it only works while using a local VSCode installation?

I'm asking that because I faced some issues (that I've already reported on Github Tickets) with GraphQL servers while using Codespaces in the browser.

I'm away for my computer right now, so I was not able to test it yet.

**Reply by 2percentsilk** [reply]
It should work with Codespaces that you access via the browser as well as ones you connect to via the local VSCode installation.

I'll take a look for your GraphQL tickets - if you remember what you were trying to do when it failed, let me know/feel free to open a new Discussion for that!

**Reply by jkimbo** [reply]
@2percentsilk I'm trying this workflow by accessing a codespace over ssh to run vim and tmux. However the container keeps stopping after some time , presumably because it's not seen any activity. Is there a way to prevent that?

**Reply by torbjornvatn** [reply]
> It should work with Codespaces that you access via the browser as well as ones you connect to via the local VSCode installation.
> 
@2percentsilk I tried to get this to work, but I wasn't able to `ssh -p2222 codespace@....` into the codespace with the `torbjornvatn-unacast-pureflow-jr5qgc5qvw.github.dev` as host. Is this described somewhere?

**Reply by BailPlus** [reply]
> > It should work with Codespaces that you access via the browser as well as ones you connect to via the local VSCode installation.
> 
> @2percentsilk I tried to get this to work, but I wasn't able to `ssh -p2222 codespace@....` into the codespace with the `torbjornvatn-unacast-pureflow-jr5qgc5qvw.github.dev` as host. Is this described somewhere?

I found the way to solve it.
1. Install vscode on your computer.
2. Open your codespaces session.
3. Click the "三" button in the corner as the following picture:
    ![image](https://user-images.githubusercontent.com/34512765/215319732-2d190a80-24f9-44f9-80c0-d75cf4ea8522.png)
4. Click the button `open in vscode desktop`
5. And the session will open on your computer soon.
6. If you have set the port forward and changed the password of user `codespace`, then execute `ssh -p 2222 codespace@localhost` And you can connect to the session with ssh.
